General/ProgrammaticCodeManipulation.md

```diff
@@ -0,0 +1,136 @@
+# Programmatic Code Manipulation (Pharo 11)
+
+In this article, we will present some code snippets that demonstrate how we can manipulate the source code (packages, classes, methods, variables, traits, etc.) programmatically, without using the SystemBrowser. Those techniques can be very useful for reflective operations, code generation, testing, etc.
+
+This article explains only those operations that **modify** the code. We will write a separate article on **querrying code** (looking for method senders, class references, pragma usages, etc.).
+
+## Packages
+
+### Creating a new package
+
+```st
+package := self packageOrganizer ensurePackage: 'PackageName'.
+```
+
+In Playground, you can use any object or class instead of `self`:
+
+```st
+package := Object packageOrganizer ensurePackage: 'PackageName'.
+```
+
+### Removing a package
+
+```st
+package removeFromSystem.
+```
+
+**Be careful!** In Pharo 11 and earlier, this will not properly remove classes from the system. So make sure to remove all classes before you remove the package.
```

Comment:
In Pharo 12 we can also do:

```st
self packageOrganizer removePackage: 'PackageName'
```

#removePackage: can take:
- A package name and remove it
- A real package and remove it
- A name of a package that does not exist and will silently do nothing in that case.

I use this a lot to create teardowns in tests because maybe the test failed directly and no package was generated. In that case I don't want the tear down to crash.